pkg/storage/utils/decomposedfs/node/locks.go

```diff
@@ -165,15 +175,6 @@ func (n *Node) RefreshLock(ctx context.Context, lock *provider.Lock) error {
 		return errtypes.PreconditionFailed("mismatching lock")
 	}
 
-	u := ctxpkg.ContextMustGetUser(ctx)
-	if !utils.UserEqual(oldLock.User, u.Id) {
-		return errtypes.PermissionDenied("cannot refresh lock of another holder")
-	}
-
-	if !utils.UserEqual(oldLock.User, lock.GetUser()) {
-		return errtypes.PermissionDenied("cannot change holder when refreshing a lock")
-	}
-
```

Comment:
This is IMHO controversal: It means a lock can be removed by anybody, right? And we need that in the context of user changes through the wopi server if I understood correctly.

Can't we get an additional flag that indicates if we allow that?

Reply:
You're right... We need to change our behaviour dependent on the lock type: https://cs3org.github.io/cs3apis/#cs3.storage.provider.v1beta1.LockType

WOPI needs to use a shared lock.

Reply:
IIRC the concept was that a lock can be refreshed either by the same `user` or by the same `appname`. Cf. the WIP for EOS: https://github.com/cs3org/reva/pull/2444/files#diff-65d95a652cc692a947b89561eaea871da49ceb5eb0b8d53ce741ae5f996b90cbR783 (which should likely be factored out).

WOPI needs to use write locks, not shared locks, otherwise we defeat the purpose of preventing sync/web clients from writing an open file.

Reply:
I think that is fine.